Index: kdebluetooth-0.4/src/trayicon.cpp
===================================================================
--- kdebluetooth-0.4.orig/src/trayicon.cpp	2010-02-01 15:37:39.000000000 -0500
+++ kdebluetooth-0.4/src/trayicon.cpp	2010-02-01 18:06:38.057388962 -0500
@@ -57,6 +57,7 @@
 #include <KApplication>
 #include <KStandardAction>
 #include <kwallet.h>
+#include <kglobalsettings.h>
 
 using KWallet::Wallet;
 KBlueTray::KBlueTray(const QString& path, QObject* parent) : KNotificationItem(parent), agentPath(path),m_aboutDialog(0)
@@ -199,13 +200,7 @@
 	confGroup = new KConfigGroup(config, "General");
 	KConfigGroup obexServerConfig(config, "ObexServer");
 	if(!obexServerConfig.hasKey("savePath")) {
-		QString saveUrl;
-		KDirSelectDialog dirs;
-		if(dirs.exec() && dirs.url().isValid()) {
-			saveUrl = dirs.url().path();
-		}else{
-			saveUrl = QDesktopServices::storageLocation(QDesktopServices::DocumentsLocation);
-		}
+		QString saveUrl=KGlobalSettings::documentPath();
 		obexServerConfig.writeEntry("savePath",saveUrl);
 		obexServerConfig.sync();
 	}
